hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -348,6 +359,44 @@ def rule_wrapper(*args, **kwargs):
     return accept
 
 
+def initialize(targets=(), target=None, **kwargs):
+    """
```

Comment:
This needs a docstring saying what it does.  Maybe something along the lines of "An initialize decorator behaves like a rule, but the decorated method is called at most once in a run. All `initialize` decorated methods will be called before any `rule` decorated methods, in an arbitrary order".

I'm not 100% happy with that. Feel free to play around with it or use your own words entirely!

On the subject of documentation, this could then also use a section in `stateful.rst` where `@invariant` and `@precondition` are already documented (embarrassingly I just realised that `@rule` does not have a section or an autodoc in that file. You could add that if you want while you're there, but you definitely don't have to do that)